src/components/DonationForm/DonationForm.jsx

```diff
@@ -1,54 +1,193 @@
+  //   control,
+  //   name: 'ImageDetail',
+  // });
+
+  // const {
+  //   fields: itemsList,
+  //   append: appendItem,
+  //   remove: removeItem,
+  // } = useFieldArray({
+  //   control,
+  //   name: 'Items',
+  // });
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [files, setFiles] = useState([]);
+  const [filesIntermediate, setFilesIntermediate] = useState([]);
+  const [descriptions, setDescriptions] = useState([]);
+  const [descriptionsIntermediate, setDescriptionsIntermediate] = useState([]);
+  // const [images, setImages] = useState([]);
+
+  const removeIntermediateFile = index => {
+    // console.log(index);
+    setFilesIntermediate(
+      filesIntermediate.filter(item => filesIntermediate.indexOf(item) !== index),
+    );
+    // console.log(filesIntermediate);
+  };
+
+  const removeIntermediateDescription = index => {
+    setDescriptionsIntermediate(
+      descriptionsIntermediate.filter(item => descriptionsIntermediate.indexOf(item) !== index),
+    );
+  };
+
+  const updateIntermediateDescription = (index, newDescription) => {
+    // console.log(newDescription);
+    console.log(descriptionsIntermediate.slice(0, index));
+    console.log(descriptionsIntermediate.slice(0, index) + [newDescription]);
+    if (index > descriptionsIntermediate.length) {
+      setDescriptionsIntermediate(descriptionsIntermediate + [newDescription]);
+    }
+    if (descriptionsIntermediate.length === 0 || descriptionsIntermediate.length === 1) {
+      console.log('Empty!');
+      setDescriptionsIntermediate([newDescription]);
+      // descriptionsIntermediate.concat([newDescription]);
+      console.log('Now: ', descriptionsIntermediate);
+    }
+    // if (descriptionsIntermediate.length === 1) {
+    //   descriptionsIntermediate
+    // }
+    if (descriptionsIntermediate.length > 1) {
+      setDescriptionsIntermediate(
+        descriptionsIntermediate.slice(0, index) + [newDescription],
+        +descriptionsIntermediate.slice(index + 1),
+      );
+    }
+    console.log(descriptionsIntermediate);
+  };
+
+  const onSubmit = async data => {
+    // const urls = await Promise.all(files.map(async file => uploadImage(file)));
+    sendEmail(data.email1, dconfirmemailtemplate);
+  };
+
+  const onOpenModal = e => {
+    setFilesIntermediate(files);
+    setDescriptionsIntermediate(descriptions);
+    onOpen(e);
+  };
+
+  const onSave = e => {
+    setFiles(filesIntermediate);
+    setDescriptions(descriptionsIntermediate);
+    console.log(descriptions);
+    onClose(e);
+  };
+
+  const onCancel = e => {
+    // helper function for closing the model
+    setFilesIntermediate(files);
+    setDescriptionsIntermediate(descriptions);
+    onClose(e);
+  };
+
+  const onSelectFurniture = (ev) => {
+    setFurnitureOptions(prev => prev.filter(e => e !== ev.target.value));
+    // document.getElementById('furnitureSelect').value = 'default';
+    appendDonation({ name: ev.target.value, num: 1 });
```

Comment:
🚫 **[eslint]** <[no-undef](https://eslint.org/docs/rules/no-undef)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'appendDonation' is not defined.